libs/utils/utils.js

```diff
@@ -724,10 +724,27 @@ async function decorateIcons(area, config) {
 async function decoratePlaceholders(area, config) {
   const el = area.querySelector('main') || area;
   const regex = /{{(.*?)}}|%7B%7B(.*?)%7D%7D/g;
-  const found = regex.test(el.innerHTML);
-  if (!found) return;
```

Comment:
Did you by any chance check whether this is quicker for areas that don't use placeholders compared to defining the tree walker? Some data on that might be interesting. Regular expressions are slow, but one is used anyway in the tree walker, so I'm curious to compare areas with and without placeholders from a performance perspective with this approach.

Reply:
> Did you by any chance check whether this is quicker for areas that don't use placeholders compared to defining the tree walker? Some data on that might be interesting. Regular expressions are slow, but one is used anyway in the tree walker, so I'm curious to compare areas with and without placeholders from a performance perspective with this approach.

Ran tests on https://placeholderreplacement--milo--suhjainadobe.hlx.live/drafts/suhjain/performance/marquee-light
Treewalker function took : 0.19999998807907104 milliseconds
async function decoratePlaceholders(area, config) { const el = area.querySelector('main') || area; const regex = /{{(.*?)}}|%7B%7B(.*?)%7D%7D/g; const walker = document.createTreeWalker( el, NodeFilter.SHOW_TEXT, { acceptNode(node) { return regex.test(node.nodeValue) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_REJECT; }, }, ); const nodes = []; // Start timing const startTime = performance.now(); let node = walker.nextNode(); while (node !== null) { nodes.push(node); node = walker.nextNode(); } // End timing const endTime = performance.now(); console.log(TreeWalker took ${endTime - startTime} milliseconds to find the nodes); if (!nodes.length) return; const { replaceText } = await import('../features/placeholders.js'); const replaceNodes = nodes.map(async (textNode) => { textNode.nodeValue = await replaceText(textNode.nodeValue, config, regex); }); await Promise.all(replaceNodes); }

Regrex function took: 0.30000001192092896 milliseconds
`async function decoratePlaceholders(area, config) {
const el = area.querySelector('main') || area;
const regex = /{{(.?)}}|%7B%7B(.?)%7D%7D/g;

// Start timing
const startTime = performance.now();
const found = regex.test(el.innerHTML);
// End timing
const endTime = performance.now();

console.log(Regex test took ${endTime - startTime} milliseconds);

if (!found) return;

const { replaceText } = await import('../features/placeholders.js');
el.innerHTML = await replaceText(el.innerHTML, config, regex);
}`

@overmyheadandbody : Please let me know if this is the data that you wanted to know?

Reply:
> Did you by any chance check whether this is quicker for areas that don't use placeholders compared to defining the tree walker? Some data on that might be interesting. Regular expressions are slow, but one is used anyway in the tree walker, so I'm curious to compare areas with and without placeholders from a performance perspective with this approach.

test